ssl/statem/statem_dtls.c

```diff
@@ -822,23 +801,23 @@ static int dtls1_read_hm_header(unsigned char *msgheaderstart,
                                 struct hm_header_st *msg_hdr)
 {
     unsigned long msg_len, frag_off, frag_len;
+    unsigned int msg_seq, msg_type;
     PACKET msgheader;
 
     if (!PACKET_buf_init(&msgheader, msgheaderstart, DTLS1_HM_HEADER_LENGTH)
-            || !PACKET_get_1(&msgheader, (unsigned int *)&msg_hdr->type)
+            || !PACKET_get_1(&msgheader, &msg_type)
             || !PACKET_get_net_3(&msgheader, &msg_len)
-            || !PACKET_get_net_2(&msgheader, (unsigned int *)&msg_hdr->seq)
+            || !PACKET_get_net_2(&msgheader, &msg_seq)
             || !PACKET_get_net_3(&msgheader, &frag_off)
             || !PACKET_get_net_3(&msgheader, &frag_len)
-            || PACKET_remaining(&msgheader) != 0
-            || msg_len > (unsigned long)SIZE_MAX
-            || frag_off > (unsigned long)SIZE_MAX
-            || frag_len > (unsigned long)SIZE_MAX) {
+            || PACKET_remaining(&msgheader) != 0) {
         return 0;
     }
 
     /* We just checked that values did not exceed max size so cast must be alright */
+    msg_hdr->type = (unsigned char)msg_type;
```

Comment:
This might be safe from a casting perspective, but is it guaranteed to be correct from an endianess perspective?  I've if you're running on a little endian system, given that this data is comming from a big endian network, will this cast loose the data you are looking for?

Reply:
@nhorman We read 1 byte from the packet to the unsigned int msg_type and then we assign it to an unsigned char value. There is no endianness issue AFAIK.

Reply:
I agree with @t8m. If it had been a pointer cast this code possibly wouldn't be correct.